lib/matplotlib/tests/test_contour.py

```diff
@@ -865,3 +865,15 @@ def test_contourf_rasterize():
     circle = mpatches.Circle([0.5, 0.5], 0.5, transform=ax.transAxes)
     cs = ax.contourf(data, clip_path=circle, rasterized=True)
     assert cs._rasterized
+
+
+@check_figures_equal(extensions=["png"])
+def test_contour_aliases(fig_test, fig_ref):
+    data = np.arange(100).reshape((10, 10)) ** 2
+    fig_test.add_subplot().contour(data, linestyle=":")
+    fig_ref.add_subplot().contour(data, linestyles="dotted")
+
+
+def test_contour_singular_color():
+    with pytest.raises(TypeError):
+        plt.figure().add_subplot().contour([[0]], color="r")
```

Comment:
The TypeError is coming from the wrong place, the first argument rather than the second.  To fix, use
`plt.figure().add_subplot().contour([[0, 1], [2, 3]], color="r")`.

Reply:
Not a blocker,  but it’s typically a good idea to use `pytest.raises(…, match=…)` to make the test more specific. This would likely have caught the issue of catching the wrong exception.

Reply:
I agree that ValueError would be more appropriate. But that doesn’t reduce the value of narrowing down tests via `match`.

Reply:
I'll leave this one as it's for now, changing pytest.raises to more generally test the error message seems like a good idea but also seems a separate issue.

Reply:
Ok.